### PR TITLE
HTTPS support for web server

### PR DESCRIPTION
**Pull request by anitakumar**
Add HTTPS support for the non grpc web server. Adds new flags to specify certificates for the webserver

### CODE REVIEW

File: cmd/matchbox/main.go [hunk 1]
```diff
@@ -47,14 +49,20 @@ func main() {
...
+	// gRPC TLS Client Authentication
+	flag.StringVar(&flags.grpcCAFile, "ca-file", "/etc/matchbox/ca.crt", "Path to the CA verify and authenticate client certificates")
 
 	// Signing
 	flag.StringVar(&flags.keyRingPath, "key-ring-path", "", "Path to a private keyring file")
 
+	// SSL flags
+	flag.StringVar(&flags.sslCertFile, "web-cert-file", "/etc/matchbox/ssl/server.crt", "Path to the server TLS certificate file")
+	flag.StringVar(&flags.sslKeyFile, "web-key-file", "/etc/matchbox/ssl/server.key", "Path to the server TLS key file")
+	flag.BoolVar(&flags.sslEnabled, "web-ssl", true, "True for set up is https")
```
Comment (olleolleolle) [inline cmd/matchbox/main.go]: Can we improve the wording of this option's comment? (`True for set up is https`)

Examples:

- True to enable HTTPS
- If True, use HTTPS

File: cmd/matchbox/main.go [hunk 2]
```diff
@@ -47,14 +49,20 @@ func main() {
...
+
+	// gRPC TLS Client Authentication
+	flag.StringVar(&flags.grpcCAFile, "ca-file", "/etc/matchbox/ca.crt", "Path to the CA verify and authenticate client certificates")
 
 	// Signing
 	flag.StringVar(&flags.keyRingPath, "key-ring-path", "", "Path to a private keyring file")
 
+	// SSL flags
+	flag.StringVar(&flags.sslCertFile, "web-cert-file", "/etc/matchbox/ssl/server.crt", "Path to the server TLS certificate file")
+	flag.StringVar(&flags.sslKeyFile, "web-key-file", "/etc/matchbox/ssl/server.key", "Path to the server TLS key file")
```
[the remark this answers is not in the document]
Comment (anitakumar) [reply cmd/matchbox/main.go]: I have incorporated your comments in the new commit

**Comment by anitakumar** [comment]
Would it possible to get this change merged in please

**Comment by sdemos** [comment]
sorry, your changes look great! I didn't merge it because I thought that the documentation I requested in my review was incoming. if not, that's totally fine, just let me know I can write some docs in a future pr and just merge this one in now. thanks!

**Comment by anitakumar** [comment]
Hi demos, 
The following is what I intend to add to the docs
# HTTPS - The matchbox API is also available with HTTPS

To start matchbox in this mode you will need the following flags set:

| Name        | Type   | Description                                                   |
|-------------|--------|---------------------------------------------------------------|
| sslEnabled  | bool   | true/false                                                    |
| sslCertFile | string | Path to the server TLS certificate file                       |
| keyFile     | string | Path to the server TLS key file                               |
| caFile      | string | Path to the CA verify and authenticate client certificates    |

I will be happy for you to add to documentation. I have not been able to login locally to push the changes for this. Please feel free to merge

**Comment by sdemos** [comment]
sorry, as a heads up I won't be able to get to this again today, I'll try and get back to it as soon as possible though, I'll make sure it's high up on my todo list. I want to run through it on my machine before merging it in.

**Comment by anitakumar** [comment]
Hi @sdemos. Did you get a chance to test this on your machine.

**Comment by dghubble** [comment]
Better to say "tls" in flags/docs and log lines (e.g. tlsCertFile), everywhere it currently says "ssl". Just bc its more bothersome to ask people to change their flags later.

**Comment by anitakumar** [comment]
> Better to say "tls" in flags/docs and log lines (e.g. tlsCertFile), everywhere it currently says "ssl". Just bc its more bothersome to ask people to change their flags later.


Thanks for your comments. We would then have to change the directories as well. Example /etc/matchbox/ssl/server.crt to /etc/matchbox/ssl/server.crt


// SSL flags
	flag.StringVar(&flags.sslCertFile, "web-cert-file", "/etc/matchbox/ssl/server.crt", "Path to the server TLS certificate file")
	flag.StringVar(&flags.sslKeyFile, "web-key-file", "/etc/matchbox/ssl/server.key", "Path to the server TLS key file")
	flag.BoolVar(&flags.sslEnabled, "web-ssl", false, "True to enable HTTPS")

File: Documentation/api.md [hunk 3]
```diff
@@ -238,3 +238,16 @@ matchbox.foo/assets/
...
 ```
+
+
+# HTTPS - The matchbox API is also available with HTTPS
+
+To start matchbox in this mode you will need the following flags set:
+
+| Name        | Type   | Description                                                   |
+|-------------|--------|---------------------------------------------------------------|
+| sslEnabled  | bool   | true/false                                                    |
```
Comment (sdemos) [inline Documentation/api.md]: all these should use the name and style that is available on the command line - `-web-ssl`, `-web-cert-file`, and `-web-key-file`, and it doesn't look like the `caFile` is actually relevant for enabling https, only for setting up the grpc tls.

**Comment by sdemos** [comment]
also, the ci errors are not caused by this patch, but they might prevent me from merging it in. they should be fixed in master though, with #723, so if you could rebase your changes onto master that would be awesome! it would also be good if you could squash your changes down to one or two commits, probably one to add the https support and another for the documentation. thanks!

**Comment by sdemos** [comment]
hi! I stlll can't merge this unless it gets rebased onto master to fix the test errors. I'll check back again tomorrow. 

I should be able to rebase it manually on my end, and if I don't see any activity on this pr tomorrow I'll see about doing that.

**Pull request merged**